float64 angle
float64 duration
---
bool ok
